.github/workflows/_publish_badge.yaml

```diff
@@ -0,0 +1,77 @@
+name: ~publish test results into a json file that can be used to make badge
+
+on:
+  workflow_call:
+    inputs:
+      SCRIPT:
+        type: string
+        description: 'Script to run :'
+        required: true
+      BADGE_LABEL:
+        type: string
+        description: 'Badge label:'
+        required: true
+      BADGE_FILENAME:
+        type: string
+        description: 'Filename to find the json in the gist for creating a badge:'
+        required: true
+      ARTIFACTS:
```

Comment:
I feel it is unnecessary to pass the artifacts using an input since we are currently downloading all job artifacts anyway. 
If we keep it as an input, we should use it in the download-artifact step to only download select artifacts.

The default value for it also seems to be specialized for pytest.

Reply:
> why not declare a variable ARTIFACT_PATTERN in the SCRIPT section and consume it locally?


That's a good point; not much utility added there. I will make that change